### ISSUE TITLE
Player not properly sitting inside of some `homedecor_seating` chairs

### ISSUE THREAD

**Issue by Lazerbeak12345** [mt-mods/homedecor_modpack#31]
When you sit on certain chairs, the player is standing, not sitting.

One such example is `lrfurn:sofa`:

![image](https://user-images.githubusercontent.com/22641188/166125230-831eafa8-0b29-4d57-b7af-77123c077efa.png)

Closes https://github.com/minetest-whynot/whynot-game/issues/84
Originally reported here: https://content.minetest.net/threads/2461/

**Comment by S-S-X** [comment]
Maybe also link to #29

**Comment by wsor4035** [comment]
this isnt a bug, this is a feature request

**Comment by wsor4035** [comment]
with https://github.com/mt-mods/homedecor_modpack/pull/28 probably the only way to do this is add per game/mod support for sitting, since thats handled by whatever player mod (such as player_api) the game is using

**Comment by wsor4035** [comment]
additionally this will be non trival, due to multiple seating options, since currently homedecor is just moving any players that right click to the nodes actual pos, no attaching or anything. so you would need to figure out which part of the couch is being pointed at, check if a player is attached there already, etc.

**Comment by Lazerbeak12345** [comment]
I'm honestly unsure how you see this as a feature request.

This also has nothing to do with making the mod game agnostic. That's a separate issue.

As for triviality, the player is already in the correct position, clearly (see photo above). So I don't see how that's related.

To keep us on topic, I'm just saying the animation needs to be the "sit" animation.

A simple way to do it would to just set their animation to the "sit" animation, and keep them in control of their rotation with camera controls. If that's too hard, then since the chair itself knows what way is forward, you can have it point there.

**Comment by Lazerbeak12345** [comment]
Re-reading, I see that you are saying that you would have to add code to check if someone is already in the chair.

Does that mean that you aren't checking for that current, and that is a problem anyway - my bug besides?

**Comment by Lazerbeak12345** [comment]
Updated OP to include link to contetdb post that started this chain of events

**Comment by Lazerbeak12345** [comment]
Dug a bit deeper.

Honestly I'm unsure how this ever got published without anyone noticing, since - as @S-S-X referenced - in #29 it was found that you can't even _sit_ on some chairs.

One could argue that this is a feature request, as it requires more code to be written, but I think that it's still a bug. Chairs are made to be sat on. How one would publish a mod in any sort of "released" state without making sure that it meets the MVP at all is beyond me.

**Comment by wsor4035** [comment]
Given how old homedecor is, it may not have been possible to have animations at the time it was written

**Comment by wsor4035** [comment]
Re trivialities, players are always put In the root node positions, for sofas this will get complicated checking which seat they are pointed at, tracking which seat position they are in, etc. Single chairs are relatively easy

**Comment by wsor4035** [comment]
Also, re this being a bug, I would define it as something being broken, causing  changes or unintended regression, etc. At the end of the day it's semantics what its labeled as, this is something that should be addressed.

**Comment by wsor4035** [comment]
Update: I have a solution for handling sofas, just now need to handle sitting animations if player api is present

**Comment by wsor4035** [comment]
i pushed my sofa code https://github.com/mt-mods/homedecor_modpack/pull/28/commits/fdb7b55a819dfefd9db50549654e0fe5c9f3068b going to need to take into account what happens when the sofa is dug from under the people in it, storing and restoring physics overrides, handling attachments, what happens if a player leaves, etc

**Comment by Lazerbeak12345** [comment]
That's a great solution. Love it so far.

**Comment by wsor4035** [comment]
fyi, thats probably as far as ill go in that pr, as i want to get it finished up today hopefully. will revisit this issue afterwords

**Comment by Lazerbeak12345** [comment]
Another example of a chair where the player is standing inside it instead of sitting is `lrfurn:armchair`

**Comment by wsor4035** [comment]
> with #28 probably the only way to do this is add per game/mod support for sitting, since thats handled by whatever player mod (such as player_api) the game is using

once https://github.com/mt-mods/xcompat/pull/41 is merged, should make game agnostic "sitting" trivial for this
